### PR TITLE
test(core): evmTransaction

### PR DESCRIPTION
**Pull request by sogunshola**
test coverate for evmTransaction
---
name: evmTransaction Test
about: A new pull request
---

### Checklist

- [X] I am not disclosing a [vulnerability](https://github.com/MoralisWeb3/Moralis-JS-SDK/blob/main/SECURITY.md).
- [X] My code is conform the [code style](https://github.com/MoralisWeb3/Moralis-JS-SDK/blob/main/CODE_STYLE.md)
- [X] I have made corresponding changes to the documentation
- [X] I have updated Typescript definitions when needed

### Issue Description

Related issue: #`FILL_THIS_OUT`

### CODE REVIEW

File: packages/core/src/test/EvmTransaction.test.ts [hunk 1]
```diff
@@ -0,0 +1,135 @@
+// Set variables
```
Comment (b4rtaz) [inline packages/core/src/test/EvmTransaction.test.ts:1]: An unnecessary comment.

File: packages/core/src/test/EvmTransaction.test.ts [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+  it('should generate eth request object with only required fields', () => {
+    const transaction = EvmTransaction.create({ chain: 1 });
+    const request = transaction.toEthRequest();
+
+    expect(request.chainId).toBe(1);
+    expect(request.value).toBeUndefined();
+  });
+
+  it('should call the send function', async () => {
```
Comment (b4rtaz) [inline packages/core/src/test/EvmTransaction.test.ts:110]: This test doesn't proof that, the callback method was called. You may do it in this way:

```
const x = jest.fn(() => { return '123'; });

x();

expect(x).toBeCalledTimes(1);
```

File: packages/core/src/test/EvmTransaction.test.ts [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+    expect(() => transaction.send()).toThrowErrorMatchingInlineSnapshot(
+      `"[C0001] Cannot send transaction, no supported call method provided"`,
+    );
+  });
+
+  it('should format EvmTransaction', () => {
+    const transaction = EvmTransaction.create(inputWithAllData);
+    const request = transaction.format();
+
+    expect(request.to).toBe(inputWithAllData.to);
```
Comment (b4rtaz) [inline packages/core/src/test/EvmTransaction.test.ts:133]: Too few expectations.

**Review by b4rtaz — CHANGES_REQUESTED** [review]

File: packages/core/src/test/EvmTransaction.test.ts [hunk 4]
```diff
@@ -111,11 +109,11 @@ describe('EvmTransaction', () => {
...
     );
+    const response = jest.fn(async () => await transactionWithSendCall.send());
 
-    const response = await transactionWithSendCall.send();
+    response();
 
-    expect(response.result.hash).toBe('0x123');
-    expect(response.result.chain?.decimal).toBe(inputWithAllData.chain);
+    expect(response).toBeCalledTimes(1);
   });
```
Comment (b4rtaz) [inline packages/core/src/test/EvmTransaction.test.ts:117]: More something like this:

```js

const callSpy = jest.fn(async (transaction) =>
    EvmTransactionResponse.create({ ...inputWithAllData, hash: '0x123', from: transaction.result.from }),
);
const transactionWithSendCall = EvmTransaction.create(inputWithAllData, callSpy);

const response = await transactionWithSendCall.send();

expect(response.result.hash).toBe('0x123');
expect(response.result.chain?.decimal).toBe(inputWithAllData.chain);
expect(callSpy).toBeCalledTimes(1);
```

In this way, you check that, the callSpy method is called exactly one time.

**Review by b4rtaz — CHANGES_REQUESTED** [review]

File: packages/core/src/test/EvmTransaction.test.ts (hunk 4 above)
Comment (sogunshola) [reply packages/core/src/test/EvmTransaction.test.ts:117]: Ahh. got it.  Thank you

**Review by b4rtaz — APPROVED** [review]

**Pull request merged**